draft-duke-quic-load-balancers.md

```diff
@@ -59,270 +61,545 @@ author:
-   MUST NOT initiate a QUIC-LB connection.
-
-   Other data MUST NOT be in the DTLS Record, so the number of
-   Connection IDs present in the packet is determined by the Record
-   length. Note that connection IDs are strings, not integers that are
-   expressed in host or network order.
-
-   A server that receives a NEW_IDS with a new CID length is likely
-   dealing with a change in load balancer configuration. It SHOULD
-   discard any unused Connection IDs in its stock and send a new
-   ID_STOCK message reflecting only Connection IDs with the new
-   length.
-
-## ID_STOCK message
-
-   Servers MUST ignore received ID_STOCK messages.
-
-     0                   1                   2                   3
-     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
-    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-    |  Type = 0x02  |
-    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-    |                    Unused Connection IDs                      |
-    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-                         Figure 2 ID_STOCK Message
-
-   This message simply reports the number of unused connection IDs in a
-   32-bit integer in Network order. Load Balancers MUST update their
-   estimate of server stock based on this message, as some connection
-   IDs may have been lost in transit.
-
-# Chained Load Balancers
-
-   In some deployments, there may be multiple tiers of trusted load
-   balancers in the path between client and server. All load balancers
-   using connection ID to encode routing information MUST agree on how
-   to decode connection IDs as routing instructions. Due to QUIC packet
-   authentication, connection IDs of established QUIC connections
-   cannot be rewritten in flight without access to the QUIC connection
-   keys.
-
-   A server configured to trust multiple load balancers MAY accept DTLS
-   connections from all of them and use provided Connection IDs
-   interchangeably. It SHOULD report its entire stock of connection IDs
-   to all trusted load balancers, rather than the number of IDs issued
-   from each source.
-
-# Security Considerations
-
+ 
+   The path between load balancer and server may pass through
+   middleboxes that could drop the coordination messages in this
+   protocol. It is therefore advantageous to make messages resemble
+   QUIC traffic as much as possible, as any viable path must obviously
+   admit QUIC traffic.
+
+## Load Balancer Chains
+
+   While it is possible to construct a scheme that supports multiple
+   low-state load balancers in the path, by using different parts of
+   the connection ID to encoding routing information for each load
+   balancer, this use case is out of scope for QUIC-LB.
+ 
+# Routing Algorithms
+ 
+   In QUIC-LB, load balancers do not send individual connection IDs to
+   servers. Instead, they communicate the parameters of one of two
```

Comment:
I would make this language generic, and not limit it to `one of two algorithms`.